### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by 1v4n4**
I created tests for the Enumerable methods. They are suited for all types of outcomes. I used let variables and made one “it” block for each expected behavior.

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+  describe '#my_select' do
+    it 'returns enumerator if no block given' do
+      expect(hash.my_select).to be_a(Enumerator)
+    end
+
+    it 'returns an array of elements for which the given block returns a true value' do
+      expect(arr.my_select { |x| x < 4 }).to eql([1, 2, 3])
+      expect(hash.my_select { |_k, v| %w[second fifth].include?(v) }).to be_a(Array)
+    end
```
Comment (ElijahTobs) [inline spec/enumerable_spec.rb]: * [x] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+  describe '#my_map' do
+    it 'returns a new array with the results of running block for every element in collection, if block given' do
+      expect(arr.my_map { |x| x * 2 }).to eql([2, 4, 6, 8, 10])
+    end
+
+    it 'if proc given, returns a new array with the results of running proc for every element in collection' do
+      expect(arr.my_map(proc { |x| x * 2 })).to eql([2, 4, 6, 8, 10])
+      expect(arr.my_map(&:to_s)).to eql(%w[1 2 3 4 5])
+    end
```
Comment (ElijahTobs) [inline spec/enumerable_spec.rb]: * [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @1v4n4 :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there are still some changes to be made before this milestone can be approved.

## HIGHLIGHTS 💯 
- Brief PR message ✔️ 
- Descriptive commit messages ✔️ 
- Kudos for adding negative scenarios ✔️ 
## REQUIRED

* [ ] Two of your specs are failing
image
* [ ] See inline comments below

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

File: spec/enumerable_spec.rb (hunk 2 above)
Comment (1v4n4) [reply spec/enumerable_spec.rb]: Done!

File: spec/enumerable_spec.rb (hunk 1 above)
Comment (1v4n4) [reply spec/enumerable_spec.rb]: Finished!

**Review by ElijahTobs — APPROVED** [review]
# Status: Approved! :tada: 

## Congratulations @1v4n4 🥂 

### Good job implementing the changes requested. 👏 

## You can now merge and submit.

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Pull request merged**